### DISCUSSION TITLE
Unable to generate JSON output from React JSX using Mitosis CLI - "no matching output target for 'builder'"

### DISCUSSION THREAD

**Discussion by johann-taberlet** [BuilderIO/jsx-lite#1560] [Q&A]
Hi everyone,

I'm currently experimenting with the Mitosis CLI to evaluate its potential for one of my side projects. Specifically, I'm trying to generate JSON output from a React JSX file but ran into an issue when specifying the output target.

Here’s the command I used:

```bash
npx mitosis compile -t builder -o ./test.json ./src/component.jsx
```
However, the CLI returns the following error:

```bash
no matching output target for "builder"
```

component.jsx:
```jsx
import { useState } from "react";

export default function MyComponent() {
	const [count, setCount] = useState(0);

	return (
		<div>
			<h1>Hello, World!</h1>
			<p>Count: {count}</p>
			<button onClick={() => setCount(count + 1)}>Increment</button>
		</div>
	);
}
```

It seems like the builder target is not recognized. Has anyone else encountered this or have any idea what might be going wrong?

I’ve checked the documentation but couldn’t find any specific guidance on this. Any help or suggestions would be greatly appreciated!

Thanks in advance!

**Comment by johann-taberlet** [comment]
While digging a bit deeper into the issue I encountered with the builder target, I took a look at the [Mitosis core source code](https://github.com/BuilderIO/mitosis/blob/d3e6511b5d75fe934b6091bfb939d2c1836f61bd/packages/core/src/targets.ts) and noticed that builder is not listed as a valid output target, even though the CLI help function includes it as a valid option:

```
 --to=<format>, -t=<format>
		Specify output format. <format> can be one of:
		
		- alpine
                - angular
                - builder
                - customComponent
                - html
                - liquid
                - lit
                - marko
                - mitosis
                - preact
                - qwik
                - react
                - reactNative
                - rsc
                - solid
                - stencil
                - svelte
                - swift
                - template
                - vue
```

As a workaround, I manually added builder to the targets in the source code, and now everything works as expected.

That said, I’m sure I might be missing something here, as I’m still a beginner compared to the awesome team who developed this CLI. Any clarification would be greatly appreciated!

Thanks a lot for your hard work and for taking the time to help!

**Reply by samijaber** [reply]
`builder` is treated a bit separately than other generators because it outputs a JSON instead of a component.

To fix the compile command, you'll need to update all uses of `targets` so that it also checks for the `builder` export coming from `@builder.io/mitosis`.
